mabwiser/rand.py

```diff
@@ -22,13 +22,13 @@ def partial_fit(self, decisions: np.ndarray, rewards: np.ndarray, contexts: np.n
 
     def predict(self, contexts: np.ndarray = None) -> Arm:
 
-        # Return a random arm
-        return self.arms[self.rng.randint(0, len(self.arms))]
+        # Return the first arm with maximum expectation
```

Comment:
to be honest, I am not sold on this design. 

This might not be written somewhere as a contract, but one of the main/core ideas behind mabwiser was that mabwiser handles the arm_to_expectations for you.. 

Right now, in some policies, the calculation is done predict() and in some policies the calculation is done in predict_exp().. 

and now I am not even sure if I take a mab model and access to its arm_to_expectations anymore (is this even tracked .e.g, here) 

The ability to have access to arm_to_exp was nice, because then we could implement CustomLPs where on predict() we overwrite it and let mab handle arm_to_expectation and we could do problem-specific algorithms/decision on top of that. 

Is it still the case that I can trust "arm_to_expectations" of any mab object?

Reply:
I don't agree with your summarization.

In this PR, for all learning policies the calculation is done in predict_expectations(), with predict() simply returning argmax(predict_expectations()).

In the current version there exists inconsistencies. For example, for Random the calculation happens in predict(), but for ThomsonSampling it happens in predict_expectations().

Re: ability to access `arm_to_expectations`. This has not been changed and can still be accessed.

What am I missing?

Reply:
GREAT to hear that arm_to_expectation remains intact. That was the main source of my confusion.

Can you help me understand this? Where does arm_to_exp is tracked? I don't see it under predict() or predict_exp()

Reply:
Ah forgot about my comment, looking into the code, I just realized that arm_to_exp is calculated in FIT and PARTIAL_FIT and has not much to do with PREDICT side of this. That's where I was confusing myself --that, we lost the arm_to_exp functionality. 

This resolves my comments.